### PR TITLE
Improvements and sane defaults for pretrain script

### PR DESCRIPTION
**Pull request by andreicozma1**

### CODE REVIEW

File: main_pretrain.py [hunk 1]
```diff
@@ -35,68 +35,91 @@ def get_args_parser():
     parser = argparse.ArgumentParser("MAE pre-training", add_help=False)
     parser.add_argument(
         "--batch_size",
-        default=64,
         type=int,
+        default=512,
```
Comment (andreicozma1) [inline main_pretrain.py:39]: Note: default batch size changed to 512

File: main_pretrain.py [hunk 2]
```diff
@@ -35,68 +35,91 @@ def get_args_parser():
...
+        type=str,
         default=None,
         choices=["group_c", "temporal", "vanilla"],
         help="Use channel model",
     )
     parser.add_argument(
         "--model",
-        default="mae_vit_base",
+        default="mae_vit_small",
         type=str,
```
Comment (andreicozma1) [inline main_pretrain.py:61]: Note: default model arch type changed to small, as even training the base is not our most common use-case for now

File: main_pretrain.py [hunk 3]
```diff
@@ -35,68 +35,91 @@ def get_args_parser():
...
+        help="The size of the square-shaped input image",
+    )
+    parser.add_argument(
+        "--patch_size",
         type=str,
-        help="attention name to use in transformer block",
+        default=8,
+        help="The size of the square-shaped patches across the image. Must be a divisor of input_size (input_size % patch_size == 0)",
     )
+
```
Comment (andreicozma1) [inline main_pretrain.py:77]: Note: default input sizes and patch sizes changed to 64 and 8, as that's the only config feasible for us

File: main_pretrain.py [hunk 4]
```diff
@@ -187,51 +217,82 @@ def get_args_parser():
...
-    parser.add_argument("--resume", default="", help="resume from checkpoint")
+
     parser.add_argument(
-        "--wandb",
+        "--resume",
         type=str,
         default=None,
-        help="Wandb project name, eg: sentinel_pretrain",
+        help="The path to the checkpoint to resume training from.",
     )
```
Comment (andreicozma1) [inline main_pretrain.py:243]: Here wandb was changed to wandb_project and moved below, but the diff looks weird

File: engine_pretrain.py [hunk 5]
```diff
@@ -25,15 +27,16 @@ def train_one_epoch(
     model.train(True)
     metric_logger = misc.MetricLogger(delimiter="  ")
     metric_logger.add_meter("lr", misc.SmoothedValue(window_size=1, fmt="{value:.6f}"))
-    header = "Epoch: [{}]".format(epoch)
+    header = f"Epoch: [{epoch}]"
```
Comment (drkostas) [inline engine_pretrain.py:30]: I'm happy to see such changes more than I should haha

File: engine_pretrain.py [hunk 6]
```diff
@@ -69,31 +72,28 @@ def train_one_epoch(
...
-            log_writer.add_scalar("train_loss", loss_value_reduce, epoch_1000x)
-            log_writer.add_scalar("lr", lr, epoch_1000x)
+            log_writer.add_scalar("train_loss", train_loss_step, epoch_1000x)
+            log_writer.add_scalar("lr", train_lr_step, epoch_1000x)
 
             # Wandb logging
-            if args.local_rank == 0 and args.wandb is not None:
-                try:
+            if args.local_rank == 0 and args.wandb_project is not None:
+                with contextlib.suppress(ValueError):
```
Comment (drkostas) [inline engine_pretrain.py:89]: Nice

**Review by drkostas — APPROVED** [review]
Nice cleanup, looks good. I'll have to adapt my train script as well after I merge

**Pull request merged**